school/diary/templates/diary/achievements_detail.html

```diff
@@ -0,0 +1,27 @@
+{#{% extends "./info_football_player.html" %}#}
+{##}
+{#{% block title %}#}
+{#    <h2>Achievements</h2>#}
+{#{% endblock title %}#}
+{##}
+{#{% block main_part %}#}
+{#    <ul>#}
+{#        {% for a in achievement_set.all %}#}
+{#            <li>{{ a.achievement }}</li>#}
+{#        {% endfor %}#}
+{#    </ul>#}
+{#{% endblock %}#}
+
+{% extends "./info_football_player.html" %}
+
+{% block title %}
+    <h2>Selected Achievement - {{ achiev }}</h2>
+{% endblock title %}
+
+{% block player_detail %}
+    <ul>
+        {% for ach in achiev.tournament_set.all %}
```

Comment:
`achiev.tournament_set.all` is incorrect since the field `tournament` is a `models. Charfield` but not a separate table connected to the Achievement by ForeignKey.

Reply:
according to your Achievement Model
```
class Achievement(models.Model):
    football_player_achievements = models.ForeignKey(FootballPlayer, on_delete=models.CASCADE)
    tournament = models.CharField(max_length=100)
    achievement = models.CharField(max_length=250)
```

when you show achievement_detail and take the specific model by the pk from the database the only data you have is  `tournament` and `achievement` 

you should show that info to the user and remove your list since there are no data for iterating on.